packages/vite/src/node/server/index.ts

```diff
@@ -129,28 +130,37 @@ export interface ServerOptions {
 }
 
 export interface ResolvedServerOptions extends ServerOptions {
-  fsServe: Required<FileSystemServeOptions>
+  fsServe: Required<Omit<FileSystemServeOptions, 'root'>>
 }
 
 export interface FileSystemServeOptions {
   /**
    * Strictly restrict file accessing outside of allowing paths.
    *
+   * Set to `false` to disable the warning
    * Default to false at this moment, will enabled by default in the future versions.
+   *
+   * @expiremental
+   * @default undefined
+   */
+  strict?: boolean | undefined
```

Comment:
**nitpick:** Is the `| undefined` explicitly needed here?

Reply:
Yes. As the value expected to be `undefined`. https://github.com/microsoft/TypeScript/issues/13195#issuecomment-832132734